daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -699,7 +699,13 @@ object NodeInfo extends Enum {
     protected sealed trait BooleanKind extends AnySimpleType.Kind
     case object Boolean extends PrimTypeNode(AnyAtomic) with BooleanKind with PrimNonNumeric with BooleanView {
       type Kind = BooleanKind
-      protected override def fromString(s: String): DataValueBool = s.toBoolean
+      protected override def fromString(s: String): DataValueBool = {
+        s match {
+          case "0" | "false" => false
+          case "1" | "true" => true
+          case _ => throw new IllegalArgumentException("Must be one of 0, 1, true, or false")
+         }
```

Comment:
Very minor, but I think there's an extra space before the curly brace so the indentation level doesn't match.

Reply:
Done